### PR TITLE
Add wmts

### LINKED ISSUE #1503: WMTS Dienst geht nicht
**Steps to reproduce**
1. Go to settings
2. Click on Map
3. Load WMTS Servis

**Expected behavior**
WMTS Service is loaded

**Current behavior**
Error Message

**Screenshots**
Image

Im WMTS-BGDI werden die in map.geo.admin.ch verfügbaren WMTS Karten angeboten. Der WMTS-BGDI ist kostenlos nutzbar. Unterstützt werden die Anfragen GetCapabilities und GetTile.

Der WMTS-BGDI ist über eine sichere Verbindung erreichbar.

In manchen Geoinformationssystemen genügt die Angabe der URL https://wmts.geo.admin.ch für die Einbindung. Ansonsten lautet der GetCapabilities-Aufruf:
(die Wahl der Sprache [de,fr,it,en] wird durch Hinzufügen des Parameters &lang=de getroffen)

MN95/CH1903+(EPSG:2056) https://wmts.geo.admin.ch/EPSG/2056/1.0.0/WMTSCapabilities.xml
MN03/CH1903 (EPSG:21781) https://wmts.geo.admin.ch/EPSG/21781/1.0.0/WMTSCapabilities.xml
Plate-Carrée WGS1984 (EPSG:4326) https://wmts.geo.admin.ch/EPSG/4326/1.0.0/WMTSCapabilities.xml
WGS84/Pseudo-Mercator (EPSG:3857, comme utilisé dans OSM, Bing, Google Maps) https://wmts.geo.admin.ch/EPSG/3857/1.0.0/WMTSCapabilities.xml

Mit dem WMS Dienst funktioniert es:

In den meisten Geoinformationssystemen genügt die Angabe der oben genannten URL für die Einbindung. Ansonsten lautet der GetCapabilities-Aufruf:
https://wms.geo.admin.ch/?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities
(die Wahl der Sprache [de,fr,it,en] wird durch Hinzufügen des Parameters &lang=de getroffen)

### PR DESCRIPTION
**Pull request by MiraGeowerkstatt**
Resolves #1503

### CODE REVIEW

File: src/client/src/pages/settings/components/editorSettingList/mapSettings.jsx [hunk 1]
```diff
@@ -1,44 +1,199 @@
...
+            wmts: wmts,
+          });
+        } else {
+          setState({
+            ...state,
+            wmsFetch: false,
+            wms: null,
+            wmts: null,
+          });
+          showAlert("Sorry, only Web Map Services (WMS) and " + "Web Map Tile Service (WMTS) are supported", "error");
```
Comment (flenny) [inline src/client/src/pages/settings/components/editorSettingList/mapSettings.jsx:185]: Q: internationalization?
PP:
```suggestion
          showAlert("Sorry, only Web Map Services (WMS) and " + "Web Map Tile Services (WMTS) are supported", "error");
```

File: src/client/src/pages/settings/components/editorSettingList/mapSettings.jsx [hunk 2]
```diff
@@ -185,117 +312,8 @@ const MapSettings = props => {
...
-                              style={{
-                                fontSize: "0.8em",
-                              }}>
-                              <Highlighter searchWords={[state.searchWmts]} textToHighlight={layer.Abstract} />
-                            </div>
-                          </div>
-                        ) : null;
-                      })}
+                  {state.wms === null ? null : getWmsList()}
+                  {state.wmts === null ? null : getWmtsList()}
```
Comment (flenny) [inline src/client/src/pages/settings/components/editorSettingList/mapSettings.jsx:316]: PP:
```suggestion
                  {state.wms && getWmsList()}
                  {state.wmts && getWmtsList()}
```

File: src/client/src/pages/settings/editorSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+          ),
+        );
+      }
+    } else if (type === "WMTS") {
+      const conf: Options | null = optionsFromCapabilities(result, {
+        layer: layer.Identifier,
+      });
+      if (conf) {
+        if (Object.prototype.hasOwnProperty.call(conf, "matrixSet") && !conf.matrixSet.includes("2056")) {
+          showAlert("Only EPSG:2056 is supported", "error");
```
Comment (flenny) [inline src/client/src/pages/settings/editorSettings.tsx:97]: Q: internationalization?
PP: duplicated code.

File: src/client/src/pages/settings/editorSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    dispatch(patchCodeConfig(`fields.${filter}`, enabled));
+  };
+  const toggleFilter = (filter: string, enabled: boolean) => {
+    dispatch(patchSettings(`efilter.${filter}`, enabled));
+  };
+  /* eslint-disable  @typescript-eslint/no-explicit-any */
+  const addExplorerMap = (layer: Layer, type: "WMS" | "WMTS", result: any, position = 0) => {
+    if (type === "WMS") {
+      if (!layer.CRS.includes("EPSG:2056")) {
+        showAlert("Only EPSG:2056 is supported", "error");
```
Comment (flenny) [inline src/client/src/pages/settings/editorSettings.tsx:70]: Q: internationalization?

**Review by flenny — DISMISSED** [review]

**Review by danjov — APPROVED** [review]

**Review by danjov — CHANGES_REQUESTED** [review]
REQ: Changelog-Eintrag

**Review by danjov — APPROVED** [review]

**Review by danjov — DISMISSED** [review]

**Review by danjov — APPROVED** [review]

**Pull request merged**